libV2/server/middleware.js

```diff
@@ -104,8 +124,8 @@ async function authV4Middleware(request, response, params) {
         throw errors.AccessDenied;
     }
 
-    if (authorizedResources !== undefined) {
-        params.body[params.resource.value] = authorizedResources;
+    if (params.level === undefined && authorizedResources !== undefined) {
```

Comment:
Just checking: is it possible that, when `level !== undefined`, `passed === true` but `authorizedResources === []` (empty array)? Wondering in case we also need to check this for authorization.

Reply:
its not, upstream `passed` is defined as `authorizedResource.length !== 0`

Reply:
this reassignment is a bit a of a hack that needs to be refactored at some point as its too specific to how the `listMetrics` handler works